Author: Sergei Golovan <sgolovan@debian.org>
Description: The patch adds a temporary workaround for a bug in kernel and
 glibc for hppa architecture where vfork()+exec() fail (see thread
 http://lists.debian.org/debian-hppa/2009/12/msg00035.html for details)
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=562218
Forwarded: no
Last-updated: Mon, 22 Feb 2010 10:34:34 +0300

--- a/erts/configure.in
+++ b/erts/configure.in
@@ -2120,6 +2120,14 @@
 AC_CHECK_DECLS([posix2time, time2posix],,,[#include <time.h>])
 
 disable_vfork=false
+AC_MSG_CHECKING([if vfork is to be disabled (for HPPA)])
+case $host_alias in
+	hppa*)
+		AC_MSG_RESULT(yes)
+		disable_vfork=true;;
+	*)
+		AC_MSG_RESULT(no);;
+esac
 if test "x$EMU_THR_LIB_NAME" != "x"; then
 	AC_MSG_CHECKING([if vfork is known to hang multithreaded applications])
 	case $host_os in
